Unity 6000.5: CS0433 metadata type conflicts with 2.1.8 and 2.1.9

## Summary
Unity CLI Loop 2.1.8 and 2.1.9 fail to compile in Unity 6000.5.0f1 with many CS0433 errors under `Editor/MetadataValidation/`.

## Observed behavior
Unity resolves both the Unity-provided `System.Reflection.Metadata` assembly and the bundled metadata assembly, then reports fully qualified type conflicts.

```text
error CS0433: The type `MetadataReader` exists in both
`System.Reflection.Metadata, Version=8.0.0.0, ...`
and `uLoopMCP.System.Reflection.Metadata, Version=8.0.0.1, ...`
```

## Suspected cause
The bundled `uLoopMCP.System.Reflection.Metadata.dll` assembly name is prefixed, but its contained types still appear under `System.Reflection.Metadata.*` namespaces. `overrideReferences: true` and narrowed `precompiledReferences` limit explicit references, but they do not appear to stop Unity 6000.5 from adding its built-in `System.Reflection.Metadata` reference. The result is a duplicate type definition at compile time.

## Compatibility notes
- Unity: `6000.5.0f1`
- Unity CLI Loop `2.0.0` through `2.1.3`: CS0619 around obsolete `GetInstanceID` APIs
- Unity CLI Loop `2.1.4` through `2.1.7`: compiles successfully
- Unity CLI Loop `2.1.8` through `2.1.9`: CS0433 conflict described above

## Workaround
Use Unity CLI Loop `2.1.7` until the metadata dependency conflict is resolved.

## Expected behavior
The package should compile on Unity 6000.5 without type conflicts between bundled metadata dependencies and Unity-provided assemblies.

## Investigation notes
Check whether the metadata dependency can be avoided for the affected editor assembly, isolated from Unity automatic references, or rebuilt/shaded so type names do not collide.

--- a/Packages/src/Cli~/src/__tests__/package-metadata.test.ts
+++ b/Packages/src/Cli~/src/__tests__/package-metadata.test.ts
@@ -28,6 +28,7 @@ const METADATA_VALIDATION_DEPENDENCY_META_PATHS = [
   'Editor/MetadataValidation/Dependencies/uLoopMCP.System.Reflection.Metadata.dll.meta',
   'Editor/MetadataValidation/Dependencies/uLoopMCP.System.Runtime.CompilerServices.Unsafe.dll.meta',
 ] as const;
+const METADATA_VALIDATION_DEPENDENCY_DEFINE_CONSTRAINT = '!UNITY_6000_5_OR_NEWER';
 const METADATA_VALIDATION_PRIVATE_ASSEMBLIES = [
   {
     relativePath:
@@ -153,6 +154,14 @@ describe('package metadata', () => {
       const metaText = loadUnityPackageText(metaPath);
 
       expect(metaText).toContain('isExplicitlyReferenced: 1');
+    }
+  });
+
+  it('excludes bundled metadata validation dependencies from Unity 6000.5 and newer', () => {
+    for (const metaPath of METADATA_VALIDATION_DEPENDENCY_META_PATHS) {
+      const metaText = loadUnityPackageText(metaPath);
+
+      expect(metaText).toContain(`- '${METADATA_VALIDATION_DEPENDENCY_DEFINE_CONSTRAINT}'`);
     }
   });
 

--- a/Packages/src/Editor/MetadataValidation/Dependencies/uLoopMCP.System.Collections.Immutable.dll.meta
+++ b/Packages/src/Editor/MetadataValidation/Dependencies/uLoopMCP.System.Collections.Immutable.dll.meta
@@ -5,7 +5,8 @@ PluginImporter:
   serializedVersion: 2
   iconMap: {}
   executionOrder: {}
-  defineConstraints: []
+  defineConstraints:
+  - '!UNITY_6000_5_OR_NEWER'
   isPreloaded: 0
   isOverridable: 1
   isExplicitlyReferenced: 1

--- a/Packages/src/Editor/MetadataValidation/Dependencies/uLoopMCP.System.Reflection.Metadata.dll.meta
+++ b/Packages/src/Editor/MetadataValidation/Dependencies/uLoopMCP.System.Reflection.Metadata.dll.meta
@@ -5,7 +5,8 @@ PluginImporter:
   serializedVersion: 2
   iconMap: {}
   executionOrder: {}
-  defineConstraints: []
+  defineConstraints:
+  - '!UNITY_6000_5_OR_NEWER'
   isPreloaded: 0
   isOverridable: 1
   isExplicitlyReferenced: 1

--- a/Packages/src/Editor/MetadataValidation/Dependencies/uLoopMCP.System.Runtime.CompilerServices.Unsafe.dll.meta
+++ b/Packages/src/Editor/MetadataValidation/Dependencies/uLoopMCP.System.Runtime.CompilerServices.Unsafe.dll.meta
@@ -5,7 +5,8 @@ PluginImporter:
   serializedVersion: 2
   iconMap: {}
   executionOrder: {}
-  defineConstraints: []
+  defineConstraints:
+  - '!UNITY_6000_5_OR_NEWER'
   isPreloaded: 0
   isOverridable: 1
   isExplicitlyReferenced: 1

